synapse/config/registration.py

```diff
@@ -432,6 +432,16 @@ def generate_config_section(self, generate_secrets=False, **kwargs):
         # users cannot be auto-joined since they do not exist.
         #
         #autocreate_auto_join_rooms: true
+
+        # Rewrite identity server URLs with a map from one URL to another. Applies to URLs
+        # provided by clients (which have https:// prepended) and those specified
+        # in `account_threepid_delegates`.
+        #
+        # Example:
+        # rewrite_identity_server_urls:
+        #   "https://somewhere.example.com": "https://somewhereelse.example.com"
+        #
+        rewrite_identity_server_urls:
```

Comment:
More of a nitpick since we're not on mainline but we usually comment out options that have default values.

Also, the fact that there's noting past the colon seems weird to me (and I would have thought that it would even make YAML parsers unhappy). We usually use a non-default example in there:

```suggestion
        #rewrite_identity_server_urls:
        #  "https://somewhere.example.com": "https://somewhereelse.example.com"
```

Reply:
Actually we do want enabled top level blocks, even if they only have comments in them: https://github.com/matrix-org/synapse/blob/master/docs/code_style.md#configuration-file-format

But I agree that it does look better if we put the example in the block, yeah.

Reply:
Well I think that relates more to config sections rather than an arbitrary dict like here. The related point even starts with:

> Where several settings are grouped into a single dict

But in this case it's only one setting, so I'd consider this case closer to e.g. the `public_baseurl` setting than to e.g. the `account_validity` section.

Reply:
Ah, ok I missed that scope declaration. Alright, I will comment the whole thing out.